src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -377,6 +385,36 @@ static ss::future<> bg_upload_and_replicate(
         co_return;
     }
 
+    if (res.value().empty()) {
+        vlog(
+          cd_log.warn,
+          "LO object upload returned empty result, nothing to replicate");
+        co_return;
+    }
+
+    auto fence_fut = co_await ss::coroutine::as_future(
+      op->ctp_stm_api->fence_epoch(res.value().front().id.epoch));
+    if (fence_fut.failed()) {
```

Comment:
ahh i see where the concurrency is that you are extracting here as there may be many inflight appends and each of these `fence_fut` instances hold a read/write lock.